chapter5/findwithflag.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func excludeNames(name string, exclude string) bool {
+	if exclude == "" {
+		return false
+	}
+	if filepath.Base(name) == exclude {
+		return true
+	}
+	return false
+}
+
+func excludeExtensions(name string, extension string) bool {
+	if extension == "" {
+		return false
+	}
+	basename := filepath.Base(name)
+	s := strings.Split(basename, ".")
+	length := len(s)
+	basenameExtension := s[length-1]
+	if basenameExtension == extension {
+		return true
+	}
+	return false
+}
+
+func main() {
+
+	minusS := flag.Bool("s", false, "Sockets")
+	minusP := flag.Bool("p", false, "Pipes")
+	minusSL := flag.Bool("sl", false, "Symbolic Links")
+	minusD := flag.Bool("d", false, "Directories")
+	minusF := flag.Bool("f", false, "Files")
+	minusX := flag.String("x", "", "Files")
+	minusEXT := flag.String("ext", "", "Extensions")
+
+	flag.Parse()
+	flags := flag.Args()
+
+	printAll := false
+	if *minusS && *minusP && *minusSL && *minusD && *minusF {
```

Comment:
this is making the logic complex, why not create a list of all and then check for each flag and remove values one by one